File: plugins/tools/sk.py
import json
import time

import requests
from pyrogram import Client, filters

from plugins.func.users_sql import *

session = requests.session()


def Getstr(string, start, end):
    string = " " + string
    ini = string.find(start)
    if ini == -1:
        return ""
    ini += len(start)
    length = string.find(end, ini) - ini
    return string[ini : ini + length].strip()


@Client.on_message(filters.command("sk"))
async def cmd_sk(Client, message):
    try:
        # NES TOOLS
        user_id = str(message.from_user.id)
        str(message.chat.type)
        str(message.chat.id)
        regdata = fetchinfo(user_id)
        results = str(regdata)
        if results == "None":
            resp = " You Are Not Registered ⚠️. First Register By Using /register To Use Me ."
            await message.reply_text(resp, message.id)
        else:
            #
            # PLAN CHECK
            await plan_expirychk(user_id)
            # PM AND AUTH CHECK
            pm = fetchinfo(user_id)
            status = pm[2]
            role = status
            if message.reply_to_message:
                sk = message.reply_to_message.text
            else:
                sk = message.text[len("/sk ") :]
            if len(sk) == 0:
                nocc = """
  𝗣𝗟𝗘𝗔𝗦𝗘 𝗣𝗥𝗢𝗩𝗜𝗗𝗘 𝗔 𝗦𝗞 𝗞𝗘𝗬 𝗧𝗢 𝗖𝗛𝗘𝗖𝗞 ⚠️
          """
                return await message.reply_text(nocc, message.id)
            else:
                chkst = "𝗖𝗵𝗲𝗰𝗸𝗶𝗻𝗴 𝗬𝗼𝘂𝗿 𝗦𝗞 𝗪𝗮𝗶𝘁...."
                done = await message.reply_text(chkst, message.id)
                tic = time.perf_counter()
                url = "https://api.stripe.com/v1/tokens"
                data = {
                    "card[number]": "4580420266153881",
                    "card[exp_month]": "09",
                    "card[exp_year]": "2025",
                    "card[cvc]": "704",
                }
                headers = {
                    "Content-Type": "application/x-www-form-urlencoded",
                }
                auth = (sk, "")

                resp = requests.post(url, data=data, headers=headers, auth=auth)
                await message.reply_text(resp, message.id)
                msg = Getstr(resp.text, '"message": "', '"')
                # BALANCE CHK

                url = "https://api.stripe.com/v1/balance"
                headers = {
                    "Content-Type": "application/x-www-form-urlencoded",
                }
                auth = (sk, "")

                response = requests.get(url, headers=headers, auth=auth)
                r2 = response.text
                parsed_data = json.loads(r2)
                await message.reply_text(r2, message.id)
                if "Expired API Key provided" in r2:
                    available_amount = "NA"
                elif "api_key_expired" in r2:
                    available_amount = "NA"
                else:
                    available_amount = parsed_data["available"][0]["amount"]

                curr = Getstr(r2, '"currency": "', '"')
                if "usd" in curr:
                    currn, currf, currs = "$", "🇺🇸", "USD"
                elif "inr" in curr:
                    currn, currf, currs = "₹", "🇮🇳", "INR"
                elif "cad" in curr:
                    currn, currf, currs = "$", "🇨🇦", "CAD"
                elif "aud" in curr:
                    currn, currf, currs = "$", "🇦🇺", "AUD"
                elif "aed" in curr:
                    currn, currf, currs = "د.إ", "🇦🇪", "AED"
                elif "sgd" in curr:
                    currn, currf, currs = "S$", "🇸🇬", "SGD"
                elif "nzd" in curr:
                    currn, currf, currs = "$", "🇳🇿", "NZD"
                elif "eur" in curr:
                    currn, currf, currs = "$", "🇪🇺", "EUR"
                elif "gbp" in curr:
                    currn, currf, currs = "£", "🇬🇧", "GBP"
                else:
                    currn, currf, currs = "N/A", "N/A", curr
                toc = time.perf_counter()
                result = f"""
  CHECK  SUCCESSFULLY 
┏－－－－－－－－－－－－┒
┠ SK - <code>{sk}</code>
┠ Resp - <code>{msg}</code>
┠ Balance - <code>{available_amount}</code>
┠ Currency - <code>{currn} {currf} {currs}</code>
┠ Time To Chk - {toc - tic:0.4f}sec
┠ Chk By - <a href="tg://user?id={message.from_user.id}"> {message.from_user.first_name}</a> [ {role} ]
┠ 𝘋𝘦𝘝 - <a href="tg://user?id=1418571871">̠print('𝙍𝝣𝘽𝝣𝙇™ </> ⚠️')</a>
┗－－－－－－－－－－－－┛
    """
                await Client.edit_message_text(message.chat.id, done.id, result)
    except Exception as e:
        print(e)

File: api/sk.php
<?php
error_reporting(0);
date_default_timezone_set('America/Buenos_Aires');

function GetStr($string, $start, $end)
{
  $string = ' ' . $string;
  $ini = strpos($string, $start);
  if ($ini == 0) return '';
  $ini += strlen($start);
  $len = strpos($string, $end, $ini) - $ini;
  return trim(strip_tags(substr($string, $ini, $len)));
}
$skval = '100';
    $sk = $_GET['sk'];


############[1 Req]#############

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, 'https://api.stripe.com/v1/tokens');
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_POSTFIELDS, "card[number]=5278540001668044&card[exp_month]=10&card[exp_year]=2024&card[cvc]=252");
curl_setopt($ch, CURLOPT_USERPWD, $sk. ':' . '');
$headers = array();
$headers[] = 'Content-Type: application/x-www-form-urlencoded';
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
$r1 = curl_exec($ch);
$msg = Getstr($r1,'"message": "','"');

############[2 Req]#############

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, 'https://api.stripe.com/v1/balance');
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_USERPWD, $sk. ':' . '');
$r2 = curl_exec($ch);
$curr = Getstr($r2,'"currency": "','"');
  $balance = trim(strip_tags(getStr($r2,'{
  "object": "balance",
  "available": [
    {
      "amount":',',')));
$pending = trim(strip_tags(getStr($r2,'"livemode": true,
  "pending": [
    {
      "amount":',',')));
if(strpos($r2,'usd')) {
  $currn = '$';
  $currf = '🇺🇸';
  $currs = 'USD';
  $pending = $pending / $skval;
  $balance = $balance / $skval;
}
elseif(strpos($r2,'inr')) {
  $currn = '₹';
  $currf = '🇮🇳';
  $currs = 'INR';
  $pending = $pending / $skval;
  $balance = $balance / $skval;
}
elseif(strpos($r2,'cad')) {
  $currn = '$';
  $currf = '🇨🇦';
  $currs = 'CAD';
  $pending = $pending / $skval;
  $balance = $balance / $skval;
}
elseif(strpos($r2,'aud')) {
  $currn = 'A$';
  $currf = '🇦🇺';
  $currs = 'AUD';
  $pending = $pending / $skval;
  $balance = $balance / $skval;
}
elseif(strpos($r2,'aed')) {
  $currn = 'د.إ';
  $currf = '🇦🇪';
  $currs = 'AED';
  $pending = $pending / $skval;
  $balance = $balance / $skval;
}
elseif(strpos($r2,'sgd')) {
  $currn = 'S$';
  $currf = '🇸🇬';
  $currs = 'SGD';
  $pending = $pending / $skval;
  $balance = $balance / $skval;
}
elseif(strpos($r2,'nzd')) {
  $currn = '$';
  $currf = '🇳🇿';
  $currs = 'NZD';
  $pending = $pending / $skval;
  $balance = $balance / $skval;
}
elseif(strpos($r2,'eur')) {
  $currn = '€';
  $currf = '🇪🇺';
  $currs = 'EUR';
  $pending = $pending / $skval;
  $balance = $balance / $skval;
}
elseif(strpos($r2,'gbp')) {
  $currn = '£';
  $currf = '🇬🇧';
  $currs = 'GBP';
  $pending = $pending / $skval;
  $balance = $balance / $skval;
}
else {
  $pending = $pending / $skval;
  $balance = $balance / $skval;
  $currn = 'N/A';
  $currf = 'N/A';
  $currs = $curr;
}
#############[Responses Balance Pending Amount  ]#############

if (strpos($r1, "rate_limit")) {
  echo "┠ Status - RATE LIMIT ⚠️<br>┠ Response - Request rate limit exceeded.<br>┠ Balance - $balance $currn<br>┠ Pending Amount - $pending $currn<br>┠ Currency - $currs $currf </b>";
}
elseif (strpos($r1, "tok")) {
  echo "┠ Status - LIVE KEY ✅<br>┠ Response - Your card was declined.<br>┠ Balance - $balance $currn<br>┠ Pending Amount - $pending $currn<br>┠ Currency - $currs $currf </b>";
}
elseif (strpos($r1, "Your card was declined.")) {
  echo "┠ Status - LIVE KEY ✅<br>┠ Response - Your card was declined.<br>┠ Balance - $balance $currn<br>┠ Pending Amount - $pending $currn<br>┠ Currency - $currs $currf </b>";
}
elseif (strpos($r1, "Invalid API Key provided")) {
  echo "┠ Status - INVALID KEY ❌<br>┠ Response - Invalid API Key Provided. </b>";
}
elseif (strpos($r1, "testmode_charges_only")) {
  echo "┠ Status - DEAD KEY ❌<br>┠ Response - Your account cannot currently make live charges. <br>┠ Balance - $balance $currn<br>┠ Pending Amount - $pending $currn<br>┠ Currency - $currs $currf </b>";
}
elseif (strpos($r1, "api_key_expired")) {
  echo "┠ Status - EXPIRED KEY ❌<br>┠ Response - Expired API Key Provided. </b>";
}
else {
  echo "┠ Status - Response NOT LISTED<br>┠ Response - $msg <br>┠ Balance - $balance $currn<br>┠ Pending Amount - $pending $currn<br>┠ Currency - $currs $currf </b>";
}

curl_close($ch);
ob_flush();
?>